packages/api-client-core/src/types.ts

```diff
@@ -74,6 +140,24 @@ type InnerSelect<Schema, Selection extends FieldSelection | null | undefined> =
         : never;
     };
 
+type InnerSelect2<Schema, Selection extends FieldSelection | null | undefined> = IfAny<
```

Comment:
The fix here is to return `never` when Selection ends up being `any`. I probably don't need this for the PR, but we should probably fix this regardless.